### PR TITLE
feat: implement send request in browse

### PR DESCRIPTION
**Pull request by kdvuong**
- while waiting for the detail page to be merge, I have put a temporary request button in item_book_list layout to test the logic and functionality.

### CODE REVIEW

**Comment by kdvuong** [comment]
demo:
Capture2
Capture3

File: app/src/main/java/com/cmput301f20t21/bookfriends/ui/browse/BrowseViewModel.java [hunk 1]
```diff
@@ -74,4 +83,14 @@ private void fetchBooks() {
...
+
+    public void sendRequest(AvailableBook book) {
+        requestRepository
+                .sendRequest(currentUsername, book.getId())
+                .addOnSuccessListener(requestId -> {
+                    int positionToUpdate = bookData.indexOf(book);
+                    book.setRequested(true);
+                    updatedPosition.setValue(positionToUpdate);
+                });
+    }
```
Comment (zhpeng811) [inline app/src/main/java/com/cmput301f20t21/bookfriends/ui/browse/BrowseViewModel.java:95]: do we need to set the status of the book to "Requested" if the previous status is "Available"?  And I think getAvailableBooksForUser() should also include a list of books with the status "Requested"

Then in the browse page we will have a list of books that have status either "Available" or "Requested", but we can still have the logic you had before, we don't display the actual book status, but display the "requested" status displayed in .setRequested() function if the specific user has requested the book

Or if it is requested by the user already it should not display in the "Browse" fragment but rather in the "Requested" tab in "Borrow" fragment? We might have to discuss about this..

Comment (kdvuong) [reply app/src/main/java/com/cmput301f20t21/bookfriends/ui/browse/BrowseViewModel.java]: updated

**Review by zhpeng811 — APPROVED** [review]
lgtm

**Review by c25vdw — APPROVED** [review]
lgtm

**Pull request merged**